### PR TITLE
Correct `pbench-fio --client` handling

### PR DESCRIPTION
**Pull request by portante**
If the user had configured a different port for a `fio` server in the `pbench-agent.cfg` file, then when clients were specified on the command line without an explicit port the default port of `8765` would be used remotely.

We also correct a debug comment with the hard-coded `8765` port.

### CODE REVIEW

**Review by webbnh — APPROVED** [review]
Assuming that the value of the `--server` is what you intend, then this seems OK by me (although, I did have a couple of comments).

File: agent/bench-scripts/tests/pbench-fio/test-05.txt [hunk 1]
```diff
@@ -204,6 +204,9 @@ fio job complete
 /var/tmp/pbench-test-bench/opt/pbench-agent/unittest-scripts/pbench-tool-meister-stop --sysinfo=default default
 /var/tmp/pbench-test-bench/opt/pbench-agent/unittest-scripts/rpm --query --queryformat=%{EVR}\n fio
 /var/tmp/pbench-test-bench/opt/pbench-agent/unittest-scripts/ssh -o BatchMode=yes -o StrictHostKeyChecking=no bar /var/tmp/pbench-test-bench/opt/pbench-agent/bench-scripts/pbench-fio --install
+/var/tmp/pbench-test-bench/opt/pbench-agent/unittest-scripts/ssh -o BatchMode=yes -o StrictHostKeyChecking=no bar cd /var/tmp/pbench-test-bench/pbench-agent/fio_test-05_1900.01.01T00.00.00/1-rw-4KiB/sample1 >/dev/null && screen -dmS fio-server bash -c '/var/tmp/pbench-test-bench/opt/pbench-agent/unittest-scripts/bm --server=,8765 2>&1 >client-result.txt'
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/bench-scripts/tests/pbench-fio/test-05.txt:207]: Yes.

Basically, if you invoke `fio --server` with no arguments, it directs `fio` to use its default port of `8765`, which also happens to be the default port that `pbench-fio` will use.

But if you change the `pbench-fio` default server port, see the `Expose client port handling bug` commit, then `pbench-fio` will not use the `--server=,####` syntax to specify the changed port.

That is what commit message, and PR description, is talking about.

File: agent/bench-scripts/pbench-fio [hunk 2]
```diff
@@ -309,11 +309,11 @@ function fio_process_options() {
...
 				client_file=$PWD/$client_file
 			fi
+			local delim=""
 			while read line; do
-				clients="$clients,$line"
+				clients="${clients}${delim}${line}"
+				delim=","
 			done < ${client_file}
-			# Remove the leading comma since $clients starts off empty
-			clients=`echo $clients | sed -e 's/^,//'`
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/bench-scripts/pbench-fio:316]: Sure, not using `sed` is a good compromise.

**Review by portante — COMMENTED** [review]
Ready for review, thanks!

**Review by dbutenhof — APPROVED** [review]

**Comment by portante** [comment]
Nothing should have changed, just rebased.

**Review by dbutenhof — APPROVED** [review]

**Review by webbnh — APPROVED** [review]
Go!

**Pull request merged**